R/utils_report.R

```diff
@@ -1,73 +1,118 @@
+#' time and test name
+#'
+#' @export
+combine_performances <- function(performance_list) {
+  # create an unique data.frame for all branches and repetitions
+  df_all <- mapply(
+    performance_list,
+    names(performance_list),
+    FUN = function(x, y) {
+      df <- bind_rows(x)
+      df$branch <- y
+
+      return(df)
+    },
+    SIMPLIFY = FALSE
+  )
+
+  # bind rows
+  df_all <- bind_rows(df_all)
+
+  # return a single data.frame
+  return(df_all)
+}
+
+#' @title Create a performance report for the tests that were run
 #'
 #' @param report_params list of tests containing commits with dates, duration
 #' time and test name
-#' @param report_name name of the file to which the report should be saved, without
-#' the extension
-#' @param report_dir name of the folder where the report should be saved
+#' @param file name of the file to which the report should be saved (.html)
 #'
 #' @importFrom quarto quarto_render
-#' @importFrom rstudioapi selectDirectory
 #' @export
-create_report <- function(report_params, report_name, report_dir) {
-  if (report_dir == "" || is.na(report_dir)) {
-    message(
-      "The name specified for the report's directory cannot be an empty string or NA. ",
-      "Make sure you're using RStudio"
+create_report <- function(report_params, file = NULL) {
+  # stop execution in case file is not provided
+  if (is.null(file)) {
+    return(
+      message("`file` cannot be NULL")
     )
-    report_dir <- selectDirectory(caption = "Please pick the report's directory")
-    if (is.null(report_dir)) {
-      return(
-        message("No directory selected. Process aborted.")
-      )
-    }
-    message(glue("The report will be automatically saved in folder {report_dir}."))
   }
 
-  file_paths <- prepare_file_paths(report_dir)
-  prepare_dir_and_template(report_dir = report_dir,
-                           file_paths = file_paths)
-  message(
-    glue(
-      "Report template was copied for you. ",
-      "You can edit and re-render it in {file_paths[2]}"
-    )
+  # manage template in order to create the report
+  report_dir <- dirname(file)
+  report_file <- basename(file)
+  report_template_file <- prepare_dir_and_template(
+    report_dir = report_dir
   )
-  message(
-    glue(
-      "You're creating a report named {report_name}. ",
-      "It'll be created in the following dir: {report_dir}"
-    )
-  )
-  message("This function is experimental!")
+  report_template_file <- basename(report_template_file)
 
-  report_file <- file.path(report_dir, glue(report_name, ".html"))
-  quarto_render(input = file_paths[2],
-                output_file = report_file,
-                execute_params = report_params)
-}
+  # generate HTML
+  # move work directory in order to run quarto
+  wb <- getwd()
+  on.exit(expr = {
+    setwd(dir = wb)
+  })
 
+  setwd(dir = report_dir)
+  quarto_render(
+    input = report_template_file,
+    output_file = report_file,
+    execute_params = report_params
+  )
+}
```

Comment:
You can consider using `withr::with_dir`, as it creates a local state (here, a local working directory) only for a part of the code, not for a whole function.

Reply:
I replaced the current code by with_dir call. It is much cleaner. Thanks!